aten/src/ATen/native/SegmentReduce.cpp

```diff
@@ -103,21 +103,114 @@ void _segment_reduce_cpu_kernel1(
         }
       });
 }
+
+template <typename T>
+void _segment_reduce_cpu_kernel2(
```

Comment:
(note for myself on question I want to ask)

This function with the adjusted striding logic would work for the *current* version of segment_reduce where only axis=0 is enabled. 

The reason being that pytorch_scatter calculates `dim` as `dim=indptr.ndim` (or in this case `lengths.ndim`) and expands `index` to match the size of the first `dim` dimensions of `src`  (or in this case `data`). As such, when `dim=0` the index can only be 1D the logic works fine.

However, if the intent of the owners of segment_reduce was to expand it to support all dimensions with a 1D index I believe this logic would fail...so should we write this as a separate kernel or unify and delete `_segment_reduce_cpu_kernel1`